conn.go

```diff
@@ -260,10 +260,14 @@ func (c *Conn) clearPendingRequests() {
 }
 
 func (c *Conn) getXid() int32 {
+	c.xidMu.Lock()
+	defer c.xidMu.Unlock()
+
 	if c.xid == math.MaxInt32 {
 		c.xid = 1
+	} else {
+		c.xid++
 	}
-	c.xid++
 
 	return c.xid
```

Comment:
An alternative to this is a per-connection goroutine generating a sequence of xids and writing to a channel. Every read from the channel would then yield an unique number: https://play.golang.org/p/3ISReQi37-K

Reply:
Do we really want a channel here?

This is only increasing a number, should we have something simpler like this?
https://github.com/go-zookeeper/zk/blob/master/conn.go#L872